### PR TITLE
RSA Encryption implementation

### PR DESCRIPTION
**Pull request by LQ95**
seems to work when tested but there are some issues and it needs improvement

### CODE REVIEW

File: src/java/com/skocur/imagecipher/encrypters/RSAEncryption.java [hunk 1]
```diff
@@ -17,30 +19,32 @@
...
-
-            byte[] encrypted = cipher.doFinal(text.getBytes());
+            byte[] encrypted;
+            if(keyformat.equals(RSAKeyType.RSA))
+            encrypted=RSATypeEncryption(text);
+            else throw new UnsupportedOperationException(); //leaving this here as a placeholder until all RSA public key formats are implemented here
 
             Encrypter encrypter = new LowLevelBitEncryption(super.fileName);
             encrypter.encrypt(new String(encrypted, StandardCharsets.UTF_8));
+            saveEncryptedData();
```
Comment (SKocur) [inline src/java/com/skocur/imagecipher/encrypters/RSAEncryption.java]: Is it necessary to save data twice? Look at LowLevelBitEncryption#encrypt(String) implementation.

Comment (LQ95) [reply src/java/com/skocur/imagecipher/encrypters/RSAEncryption.java]: Ah,yeah I didn't really notice that. maybe I should remove that last line?

Comment (SKocur) [reply src/java/com/skocur/imagecipher/encrypters/RSAEncryption.java]: Please remove and commit new changes to this request.

Comment (LQ95) [reply src/java/com/skocur/imagecipher/encrypters/RSAEncryption.java]: Done

**Pull request merged**